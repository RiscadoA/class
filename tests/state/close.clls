proc main() {
    cut {
        println("cell 1"); cell c1(x. affine x; println("close 1"); close x)
        |c1: ~state close|
        println("take 1"); take c1(x);
        println("wait 1"); wait x;
        fwd c1 e
        |e: ~statel close|
        println("put 1"); put e(x. affine x; println("close 2"); close x);
        fwd e c2
        |c2: ~state close|
        println("take 2"); take c2(x);
        println("wait 2"); wait x;
        println("put 2"); put c2(x. affine x; println("close 3"); close x);
        println("release"); release c2
    }
};;
