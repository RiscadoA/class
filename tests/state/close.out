cell 1
close 1
take 1
wait 1
put 1
close 2
take 2
wait 2
put 2
close 3
release
